Alameen688/MyDiary
# #159465173 Frontend Client should be able to perform add/edit diary entry by connecting to server

#### What does this PR do?
Implements a working frontend add and edit entry by connecting ui template and server api
endpoint for adding and editing of diary entries. A change to the
auth.js file is also included to display error message in catch block to
the user (based on LFA feedback)
#### Description of Task to be completed?
The following should be working
- Add new diary entry
- Edit diary entry
#### How should this be manually tested?
After cloning the repo, cd into it then
- make sure you have postgresql installed on your machine and start it.
- add a postgreSQL DATABASE_URL to your .env file
   ```
     DATABASE_URL=postgresql://username:password@localhost:5432/mydiarydb
   ```
 - RUN  `npm run dev` to start the server
 - Navigate to `/client/add-entry.html`, you should be able to create a new entry and also receive validation errors on using invalid inputs 
 - Navigate to `/client/edit-entry.html?id=<entry_id>`, you should be edit the entry with the given id created by the logged in user OR receive validation errors on using invalid inputs as the case might be
#### Any background context you want to provide?
none
#### What are the relevant pivotal tracker stories?
[#159465173](https://www.pivotaltracker.com/story/show/159465173)
